### PR TITLE
Add evaluations for od and ic

### PR DESCRIPTION
**Pull request by pj-ms**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+setuptools.setup(name='vision-evaluation',
+                 author='Ping Jin, Shohei Ono, I-Ting Fang',
+                 description="Evaluation code for vision tasks.",
+                 long_description=long_description,
+                 long_description_content_type="text/markdown",
+                 url='https://github.com/pj-ms/vision-evaluation',
+                 version=VERSION,
+                 license='MIT',
+                 python_requires='>=3.5',
```
Comment (shonohs) [inline setup.py:16]: this code doesn't support python3.5. f-string is introduced in 3.6

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import setuptools
+
+VERSION = '0.1.0'
+
+with open("README.md", "r") as fh:
```
Comment (shonohs) [inline setup.py:5]: this can fail if the working directory is different. use __ file __ to get the dir?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+# Introduction 
+This repo contains evaluation metric codes used in Microsoft Cognitive Services Computer Vision for tasks such as classification and object detection.
```
Comment (shonohs) [inline README.md:2]: Need a document about API usage

File: vision_evaluation/evaluators.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
+import collections
+import statistics
+import sklearn.metrics
+import numpy as np
+from abc import ABC
+
+
+def top_k_prediction_indices(prediction, k):
```
Comment (shonohs) [inline vision_evaluation/evaluators.py:8]: add '_' prefix if this function is not intended to be used by library user?

File: vision_evaluation/evaluators.py [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+    def __init__(self, ious=DEFAULT_IOU_VALUES, random_seed=0):
+        self.evaluators = [MeanAveragePrecisionEvaluatorForSingleIOU(random_seed, iou)
+                           for iou in ious]
+        super(MeanAveragePrecisionEvaluatorForMultipleIOUs, self).__init__(random_seed)
+
+    def add_predictions(self, predictions, targets):
+        for evaluator in self.evaluators:
+            evaluator.add_predictions(predictions, targets)
+
+    def get_report(self, average='macro'):
```
Comment (shonohs) [inline vision_evaluation/evaluators.py:326]: I feel this argument should be in the constructor.

File: README.md (hunk 3 above)
Comment (pj-ms) [reply README.md:2]: sure

File: vision_evaluation/evaluators.py (hunk 5 above)
Comment (pj-ms) [reply vision_evaluation/evaluators.py:326]: I forgot to change it to kwargs. The thing is, if user wants get perf of two kinds of average, a new evaluator instance has to be created and add_predictions have to be called all over again.

Comment (shonohs) [reply vision_evaluation/evaluators.py]: I guess, when user use multiple evaluators, they would like to do like:
evalutors = [Evaluator1(...), Evaluator2(...), ...]

for e in evaluators:
....e.add_predictions(...)

report = {k, v for e in evaluators for k, v in e.get_report().items()}

if we need to give evaluator-specific argument in get_report(), we will limit this kind of usage. Can we at least let user set default 'average' setting in the constructor?

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+                 version=VERSION,
+                 license='MIT',
+                 python_requires='>=3.6',
+                 packages=setuptools.find_packages(),
+                 keywords='vision metric evaluation classification detection',
+                 classifiers=[
+                     'Development Status :: 4 - Beta',
+                     'Intended Audience :: Developers',
+                     'License :: OSI Approved :: MIT License',
+                     'Programming Language :: Python :: 3.5',
```
Comment (shonohs) [inline setup.py:27]: Remove 3.5 tag

File: vision_evaluation/evaluators.py [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+
+    return target_mat
+
+
+class Evaluator(ABC):
+    """Class to evaluate model outputs and report the result.
+    """
+
+    def __init__(self, random_seed=0):
+        self.random_seed = random_seed
```
Comment (shonohs) [inline vision_evaluation/evaluators.py:29]: what is this random_seed for?

Comment (pj-ms) [reply vision_evaluation/evaluators.py:29]: copied from iristorch

Comment (shonohs) [reply vision_evaluation/evaluators.py]: Tracked down the change, and it turned out it was me added this property. However, the corresponding feature is already removed, so please remove this parameter lol

**Review by shonohs — APPROVED** [review]
looks good to me

**Pull request merged**